src/ripple/protocol/impl/BuildInfo.cpp

```diff
@@ -110,18 +110,15 @@ getVersionString ()
     struct SanityChecker
     {
         SanityChecker ()
+            : versionString (getRawVersionString ())
         {
             beast::SemanticVersion v;
 
-            char const* const rawText = getRawVersionString ();
-
-            if (! v.parse (rawText) || v.print () != rawText)
-                beast::FatalError ("Bad server version string", __FILE__, __LINE__);
-
-            versionString = rawText;
+            if (! v.parse (versionString) || v.print () != versionString)
+                LogicError (versionString + ": Bad server version string");
```

Comment:
Since you're cleaning this up, how about getting rid of the local struct and using a lambda. Something like:
```
std::string const&
getVersionString ()
{
    static std::string const value = [] {
        std::string const versionString = getRawVersionString ();
        beast::SemanticVersion v;
        if (!v.parse (versionString) || v.print () != versionString)
            LogicError (versionString + ": Bad server version string");
        return versionString;
    }();
    return value;
}
```